### ISSUE TITLE
The performance of the sample code is different when running on multiple GPUs versus a single GPU.

### ISSUE THREAD

**Issue by pengyrDL** [huggingface/diffusers#4468]
### Describe the bug

I tried to train unconditional diffusion with accelerate on a dataset generated from .ttf files.  And I find that the performance of the sample code is different when running on multiple GPUs versus a single GPU.

I trained on two A100 GPUs with a learning rate of 1e-4 and batch size of 32, and found the results to be much better than training on a single A100 GPU with a learning rate of 1e-4 and batch size of 64.

First I noticed that in the code, each process calls lr_scheduler.step(), causing the learning rate to differ. But even after fixing the discrepancy, the generation quality was still very poor.

With multiple GPUs, after 800 epochs I could generate pretty decent images, but with a single GPU and 64 batch size, even at 3000 epochs the image quality was still very bad.

### Reproduction

accelerate launch --gpu_ids=0  --num_processes=1  train_unconditional.py --train_batch_size 64 --checkpointing_steps 1000 --save_model_epochs 500 --learning_rate 1e-4 --train_data_dir ./datasets/fontimages/ --output_dir ./result_uncondition_1e4_64 --ddpm_num_inference_steps 20 --num_epochs 5000

accelerate launch --multi_gpu --gpu_ids=0,1  --num_processes=2 train_unconditional.py --train_batch_size 32 --checkpointing_steps 1000 --save_model_epochs 500 --learning_rate 1e-4 --train_data_dir ./datasets/fontimages/ --output_dir ./result_uncondition_1e4_32_multigpu  --ddpm_num_inference_steps 20 --num_epochs 5000

### System Info

- `diffusers` version: 0.19.3
- Platform: Linux-5.15.0-76-generic-x86_64-with-glibc2.35
- Python version: 3.9.17
- PyTorch version (GPU?): 2.0.1 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.31.0
- Accelerate version: 0.21.0
- xFormers version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by pengyrDL** [comment]
![BVK_2YD@IMN_V$KU7KO{Z4L](https://github.com/huggingface/diffusers/assets/138228198/6d643754-354a-4acc-ad27-25cd96e088e4)
